src/pages/api/v1/image-upload/multipart/index.ts

```diff
@@ -0,0 +1,40 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getServerAuthSession } from '~/server/utils/get-server-auth-session';
+import { getMultipartPutUrl, getS3Client } from '~/utils/s3-utils';
+import { env } from '~/env/server.mjs';
+import { randomUUID } from 'crypto';
+
+const s3Domain = (env.S3_IMAGE_UPLOAD_ENDPOINT ?? env.S3_UPLOAD_ENDPOINT).replace(
+  /https?\:\/\//,
+  ''
+);
+
+export default async function imageUploadMultipart(req: NextApiRequest, res: NextApiResponse) {
+  const session = await getServerAuthSession({ req, res });
+  const userId = session?.user?.id;
+  if (!userId || session.user?.bannedAt) {
```

Comment:
Maybe throw in the mix muted as well?
```suggestion
  if (!userId || session.user?.bannedAt || session.user?.muted) {
```